ghost/core/core/server/services/link-redirection/LinkRedirectRepository.js

```diff
@@ -31,6 +31,22 @@ module.exports = class LinkRedirectRepository {
         linkRedirect.link_id = ObjectID.createFromHexString(model.id);
     }
 
+    async getAll(options) {
+        const collection = await this.#LinkRedirect.findAll(options);
+
+        const result = [];
+
+        for (const model of collection.models) {
+            result.push(new LinkRedirect({
+                id: model.id,
+                from: new URL(model.get('from'), this.#urlUtils.getSiteUrl()),
+                to: new URL(this.#urlUtils.transformReadyToAbsolute(model.get('to')))
```

Comment:
`transformReadyToAbsolute` has moved to the model layer for now because the activity feed also needs to return these. Ideally the activity feed would also use the repository pattern, but with all the relations that would become difficult (because we need click event models, member models, link models and post models) unless we just ignore the whole bookshelf relations and start wiring up the relations manually using repositories. Then do a call to the LinkClickRepository to get the 'events' (and where you need to pass in a repository for members, links and posts), which returns a different kind of object that includes the member, link and posts. Curious to know your opinion on this.

Reply:
Ah cool, good to know, thanks!

The plan is to still keep all of the DB code in the models for now, so that everything is in one place for pulling out/improving/replacing in future. I think the repository pattern serves us best in new code so we can easily stub and mock the data layer in development, as well as typing all of our code, we don't need to update the activity feed stuff just yet I don't think :)